### ISSUE TITLE
1.4 API change not usable?

### ISSUE THREAD

**Issue by brianpeiris** [adngdb/entity-system-js#29]
Not sure if you still care to maintain this library. I've noticed that the API change introduced in v1.4.0 makes it so you can't easily get the id of entities unless you access `entityComponentData` directly, which I'm not sure is the intended use. For example the "concentration" example made heavy use of entity ids retrieved from the previous version of `getComponentsData`.

What was the reason for the change? What is the recommended method of querying entities and their ids?

**Comment by brianpeiris** [comment]
Also, should the `entityComponentData` structure be considered part of the public API?

**Comment by adngdb** [comment]
Hey @brianpeiris, thanks for reporting this! I am happy to keep maintaining this library, but I am not using it those days, nor do I intend to do so in the foreseeable future. If you're using it though, I'd be delighted to know how and where. :-)

I did not realize that this change was the change in 1.4.0 was this big. You can still access the ID of an entity by using `.__id` on the objects returned by `getComponentsData`. However, after having just updated the concentration example, it is indeed not very convenient. As I'm not using this tool anymore, I'd rather do what actual users would prefer. So if you have an opinion or a favorite resolution for this, please let me know. 

`entityComponentData` should not theoretically be part of the public API, even though right now it *de facto* is. 

*Side note: I've been using redux a lot over the last few years and I think it is basically an entity system but better.*

**Comment by brianpeiris** [comment]
Thanks for the response. I'm not actively using ensy, but I did include it https://TowerDefenseECS.com/ as a comparison to other ECS libraries, which is why I stumbled upon this API change. I ended up just using v1.3 instead since it would have been very awkward without access to entity ids.
https://github.com/brianpeiris/TowerDefenseECS/blob/master/src/libraries/ensy/main.js

If you would like people to continue using the library, it seems the right thing to do would be to release a v1.5 with the API reverted back to what it was in v1.3.
